Shopify/cli
# Add pos_ui_extension to migration for ui_extension

### WHY are these changes introduced?

We noticed that while testing a deploy for Stocky to the new ui extension spec, that the registrationUuid gets regenerated. This is problematic for smart grid tiles that are already existing, as they refer to these uuids in order to exist. If the registrationUuid no longer exists, then the tile will break

### WHAT is this pull request doing?

Adds pos_ui_extension extension types to the migrator.

### How to test your changes?

1. I have a separate branch that has a test extension commited. Check this branch out in my [spin instance](https://extensions-hu29.jeansebastien-goupil.us.spin.dev). 
2. You'll need to create a new app/extension with the old spec. You can just copy this folder into your local spin instance. Make sure you run npm install to install the dependencies.
3. Deploy the extension by doing `pnpm shopify app deploy --path {path to your app} --verbose`. This should deploy the extension in the old spec.
4. For convenience purposes, attach the app to the existing Extension Spin app, since it's already published on shop 1.
5. Go to graphQL for shop-1, and query with the `specificationIdentifier: "pos_ui_extension"` to make sure that it's there. Take note of the registrationUuid.
6. Change the toml file to the following: 

```
name = "sandbox-170"
type = "ui_extension"
handle = "sandbox-170"
description = "sandbox-170"
api_version = "2024-04"

[[extension_points]]
target = "pos.home.tile.render"
module = "./src/index.tsx"

```

8. Run `pnpm build` and `pnpm shopify app build`. 
9. Run `pnpm shopify app deploy --path {path to your app} --verbose` again. This time you should get a prompt to migrate your extension. Hit yes.
10. This should run succesfully, and prompt you to deploy a new version.
11. Once that's done, go to graphql for shop-1 and query with the `specificationIdentifier: "ui_extension"`. You should see your extension there with the SAME `registrationUuid`.

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
